sentry_streams/sentry_streams/adapters/arroyo/reduce.py

```diff
@@ -66,6 +75,64 @@ def accumulator(self, result: Any, value: BaseValue[RoutedValue]) -> RoutedValue
         return routed
 
 
+class MetricsAccumulator(Accumulator[Message[Any], Any]):
```

Comment:
Unit tests please also. You can use a Dummy metrics backend that tracks what we produce like in snuba.